[Lyu Xingyu] iP

Zemdalk opened the pull request.

Zemdalk committed:
Level-0: initial skeletal version of the Duke that simply greets the user and exits.

Zemdalk committed:
Level-1: Greet, Echo, Exit

Zemdalk committed:
Level-2: Add, List

Zemdalk committed:
Level-3: Mark as Done

Zemdalk committed:
A-CodingStandard

Zemdalk committed:
Level-4: ToDos, Events, Deadlines

Zemdalk committed:
A-TextUiTesting

Zemdalk committed:
A-TextUiTesting

Zemdalk committed:
A-CodeQuality: adjust indent and delete some useless comments

lihka1202 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,102 @@
+        while(true){
+            String line = in.nextLine();
+            System.out.println(line);
+            if(line.equals("bye")){
+                // quit
+                System.out.println(exitPrompt);
+                break;
+            }else{
```
Perhaps you can leave a space after the conditional clause and opening brace

lihka1202 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,102 @@
+                }else if(func.equals("unmark")){
+                    // mark work as unfinished
+                    int index;
+                    try {                        
+                        index = Integer.parseInt(line.substring(funcIdx+1, line.length())) - 1;
+                    } catch (Exception e) {
+                        index = -1;
+                    }
```
I like how you have added error handling

TopGun2001 replied:
New implementation I've noticed. I have not seen this before. Maybe something that I should google up. Well done~

lihka1202 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,102 @@
+                if(funcIdx == -1){
+                    funcIdx = line.length();
+                }
+                String func = line.substring(0, funcIdx);
```
perhaps you could give this variable a better name, as it is stored as a `String` but is named `func`

TopGun2001 replied:
I agree with @lihka1202 . I'd suggest to give a variable name like **commandIndex** or **inputIndex** as it it a bit more clear for readers to understand what you are referring to.

Zemdalk replied:
Thank you for your good suggestions! I have made corrections.

lihka1202 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,102 @@
+                    }
+                    todoList.markItem(index, false);
+                }else if(func.equals("todo")){
+                    Todo todo = Todo.toTodo(instruction);
```
Perhaps you could name object `todo` better, as it has the same name as the class and it might get confusing!

lihka1202 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,102 @@
+                    // mark work as done
+                    int index;
+                    try {                        
+                        index = Integer.parseInt(line.substring(funcIdx+1, line.length())) - 1;
```
Could you abstract this out to maintain SLAP

Zemdalk replied:
Agreed. Thank you for your suggestion!

TopGun2001 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,28 @@
+    protected String to;
+
+    public static Event toEvent(String instruction){
+        int contentIdx = instruction.indexOf("/from");
+        int fromIdx = instruction.indexOf("/to");
```
Perhaps u could use **Index** instead of **Idx**

0nandon commented on src/main/java/Duke.java:
```diff
@@ -51,14 +66,31 @@ public static void main(String[] args) {
                     todoList.markItem(index, false);
+                }else if(func.equals("todo")){
+                    Todo todo = Todo.toTodo(instruction);
+                    if(todo == null){
+                        printError("Wrong todo format");
+                        continue;
+                    }
+                    todoList.addItem(todo);
+                }else if(func.equals("deadline")){
+                    Deadline deadline = Deadline.toDeadline(instruction);
+                    if(deadline == null){
+                        printError("Wrong deadline format");
+                        continue;
+                    }
+                    todoList.addItem(deadline);
+                }else if(func.equals("event")){
+                    Event event = Event.toEvent(instruction);
+                    if(event == null){
+                        printError("Wrong event format");
+                        continue;
+                    }
+                    todoList.addItem(Event.toEvent(instruction));
```
'switch' statement would be better since there are many cases you want to separate.

0nandon replied:
There is deep nesting in **if-else** statement while every **if** statements inside are conducting similar calculation. There would be other way to better construct this code like declaring special function for those calculation.

0nandon commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+        return new Deadline(deadlineContent, deadlineBy);
+    }
+
+    public Deadline(String description, String by) {
```
```suggestion
public getDeadline(String description, String by) {
```

0nandon reviewed (commented):
Function is better to be in verb form.

0nandon reviewed (commented):
It is better to avoid deep nesting as possible as you can.

0nandon commented on src/main/java/TodoList.java:
```diff
@@ -12,51 +12,49 @@ public int getListnum(){
             }
             System.out.println(tasks[num].getDescription());
             System.out.println(SPLITTER);
             System.out.println();
```
So many printings, I think in if-else statement. How about just making additional function for those printing stacks?

Zemdalk committed:
code: follow some PRs' suggestions

Zemdalk committed:
code: follow some PRs' suggestions

Zemdalk committed:
Level-5: add exception handlers

Zemdalk committed:
Level-5: remove leading and trailing zeros in exception handling

Zemdalk committed:
Merge branch 'branch-Level-5'

Zemdalk committed:
A-Packages

Zemdalk committed:
Level-6: Delete

Zemdalk committed:
Level-7: Save

Zemdalk committed:
Level-7: Save

Zemdalk committed:
A-Jar

Zemdalk committed:
refactor: modify code according to TA's suggetions

Zemdalk committed:
file: add  in

Zemdalk committed:
A-MoreOOP

Zemdalk committed:
Level-8: Dates and Times

Zemdalk committed:
Merge pull request #4 from Zemdalk/branch-Level-8

Level-8: Dates and Times

Zemdalk committed:
Level-9: Find

Zemdalk committed:
Merge pull request #5 from Zemdalk/branch-Level-9

Level-9: Find

Zemdalk committed:
A-JavaDoc

Add header comments to all non-private classes/methods, and non-trivial private methods.

Zemdalk committed:
A-UserGuide: add a user guide

Zemdalk committed:
A-UserGuide: add a user guide

Zemdalk committed:
A-UserGuide: add a user guide
